src/bam/record.rs

```diff
@@ -1126,43 +1123,54 @@ impl TryFrom<&str> for CigarString {
 
     /// Create a CigarString from given str.
     fn try_from(text: &str) -> Result<Self> {
-        lazy_static! {
-            // regex for a cigar string operation
-            static ref OP_RE: Regex = Regex::new("^(?P<n>[0-9]+)(?P<op>[MIDNSHP=X])").unwrap();
-        }
+        let bytes = text.as_bytes();
         let mut inner = Vec::new();
         let mut i = 0;
-        while i < text.len() {
-            if let Some(caps) = OP_RE.captures(&text[i..]) {
-                let n = &caps["n"];
-                let op = &caps["op"];
-                i += n.len() + op.len();
-                let n = u32::from_str(n).map_err(|_| Error::BamParseCigar {
-                    msg: "expected integer".to_owned(),
-                })?;
-                inner.push(match op {
-                    "M" => Cigar::Match(n),
-                    "I" => Cigar::Ins(n),
-                    "D" => Cigar::Del(n),
-                    "N" => Cigar::RefSkip(n),
-                    "H" => Cigar::HardClip(n),
-                    "S" => Cigar::SoftClip(n),
-                    "P" => Cigar::Pad(n),
-                    "=" => Cigar::Equal(n),
-                    "X" => Cigar::Diff(n),
-                    op => {
-                        return Err(Error::BamParseCigar {
-                            msg: format!("operation {} not expected", op),
-                        });
-                    }
-                });
-            } else {
+        let text_len = text.len();
+        while i < text_len {
+            let mut j = i;
+            while j < text_len && bytes[j].is_ascii_digit() {
+                j += 1;
+            }
+            // check that length is provided
+            if i == j {
                 return Err(Error::BamParseCigar {
-                    msg: "expected cigar operation [0-9]+[MIDNSHP=X]".to_owned(),
+                    msg: "Expected length before cigar operation [0-9]+[MIDNSHP=X]".to_owned(),
                 });
             }
+            // get the length of the operation
+            let n = u32::from_str(&text[i..j]).map_err(|_| Error::BamParseCigar {
+                msg: "Unable to parse to u32".to_owned(),
```

Comment:
Maybe make this more specific to ease debugging, something like:
```suggestion
                msg: format!("Unable to parse &str '{}' to u32.", &test[i..j]).to_owned(),
```